brave/brave-core
# Add metrics for NTP search widget

Resolves https://github.com/brave/brave-browser/issues/39762



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start with fresh profile, view the NTP.
1. Ensure `Brave.Search.WidgetDefault` has value 0.
2. Change the default engine in the widget, and ensure that the metric has the correct value.
3. Disable the widget, ensure the metric no longer exists
4. Enable the widget, ensure the `Brave.Search.WidgetUsage` metric does not exist and the `Brave.Search.WidgetDefault` metric has the correct value.
5. Make queries using the widget. Ensure the `Brave.Search.WidgetUsage` metric has the correct value.
